docs/design/design.md

```diff
@@ -0,0 +1,116 @@
+* MicroShift can use both user-local and system-wide configuration.
+
+### Security
+* MicroShift instances should run with least privileges. In particular Control Plane-only instances should run completely non-privileged.
+* MicroShift should minimize the number of open network ports. In particular Node-only instances should not open any listening ports.
+* Assume there is no way to access the device or MicroShift instance from remote, i.e. no SSH access nor Kubernetes API access (for kubectl). Instead, communication is *always* initiated from the device / cluster towards the management system.
+* Open issues / questions:
+  * Model for joining nodes to MicroShift clusters.
+  * Model for certificate rotation.
+  * Requirements for FedRAMP and DISA STIG compliance.
+  * How to leverage / support the host OS's features for FIDO Device Onboard, remote attestation, integrity measurement architecture.
+  * How to support secure supply chain.
+
+### Networking
+* Host networking is configured by device management. MicroShift has to work with what it's been given by the host OS.
+* No Multus.
+* Open issues / questions:
+  * Lightweight CNI?
+  * API Load Balancing?
+  * Service Load Balancing?
+  * Ingress?
+
+### Storage
+* MicroShift defaults to local ephemeral storage (enough for basic use cases).
+* Open issues / questions:
+  * Provide escape hatch to add own CSI (which?).
+
+### Production / Supply Chain / Release Management
+* MicroShift vendors OKD source code without modification. Where it deploys container images for additional services, it deploys OKD's published container images, not the OpenShift downstream's.
+* MicroShift's versioning scheme follows OKD's. This scheme signals the base OpenShift version (4.x) and order/age of builds, but intentionally avoids signaling patch level, backward compatibility (as SemVer, for example), or stability.
+* We ensure the tip of our development branch is deployable and while MicroShift is still early days and experimental we expect developers (and users who want the "latest") to build & deploy from source.
+* Releases are mainly provided for convenience to useres that just want to give MicroShift a quick try without friction. They are cut irregularly, e.g. to make a new feature available.
+* When rebasing onto a new OKD version, we vendor that version's packages and update the container image digests of the infrastructure services MicroShift deploys, i.e. the "release metadata" is baked into the MicroShift binary.
+* Eventually, we expect there to be a "MicroShift Release Image" that is based on / derived from the OpenShift Release Image: It references the MicroShift container image plus the subset of container images shared with and published by OpenShift. Defining a release image should allow to reuse the proven OpenShift CI and release toolinng later.
```

Comment:
```suggestion
* Eventually, we expect there to be a "MicroShift Release Image" that is based on / derived from the OpenShift Release Image: It references the MicroShift container image plus the subset of container images shared with and published by OpenShift. Defining a release image should allow to reuse the proven OpenShift CI and release tooling later.
```
don't mind me :smile_cat: just browsing through

Reply:
I'll happily take that :). Fixed, thanks.